W3ChampionsIdentificationService/RolesAndPermissions/CommandHandlers/UsersCommandHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using W3ChampionsIdentificationService.Middleware;
+using W3ChampionsIdentificationService.RolesAndPermissions.Contracts;
+
+namespace W3ChampionsIdentificationService.RolesAndPermissions.CommandHandlers
+{
+    public class UsersCommandHandler : IUsersCommandHandler
+    {
+        private readonly IUsersRepository _usersRepository;
+        private readonly IRolesRepository _rolesRepository;
+        private readonly RolesAndPermissionsValidator _validator;
+        public UsersCommandHandler(
+            IUsersRepository usersRepository,
+            IRolesRepository rolesRepository,
+            RolesAndPermissionsValidator validator)
+        {
+            _usersRepository = usersRepository;
+            _rolesRepository = rolesRepository;
+            _validator = validator;
+        }
+
+        public async Task CreateUser(UserDTO user)
+        {
+            await _validator.ValidateRoleListHttp(user.Roles);
+            await _validator.ValidateUserDtoHttp(user);
+            var allUsers = await _usersRepository.GetAllUsers();
+            var roles = await _rolesRepository.GetAllRoles(x => user.Roles.Contains(x.Id));
+
+            var distinctPermissions = roles

```

Comment:
Why do you need to set user permissions here? They should be retrieved from roles when token is created and added to the token.